src/ui/libs/DatalensChartkit/ChartKit/plugins/Table/renderer/components/Table/BackgroundTable.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+
+import {Portal} from '@gravity-ui/uikit';
+import block from 'bem-cn-lite';
+
+import {COMPONENT_CLASSNAME} from '../../../../../../../../components/Widgets/Chart/helpers/helpers';
+import {waitForContent} from '../../../../../helpers/wait-for-content';
+import type {WidgetDimensions} from '../../types';
+
+import {TableBody} from './TableBody';
+import {TableFooter} from './TableFooter';
+import {TableHead} from './TableHead';
+import type {TableViewData} from './types';
+import {getTableSizes} from './utils';
+
+const b = block('dl-table');
+
+type Props = {
+    dimensions: WidgetDimensions;
+    data: TableViewData;
+    onChangeMinWidth?: (cellSizes: number[]) => void;
+};
+
+export const BackgroundTable = React.memo<Props>((props: Props) => {
+    const {
+        dimensions,
+        data: {header, body, footer},
+        onChangeMinWidth,
+    } = props;
+
+    const containerRef = React.useRef<HTMLDivElement | null>(null);
+    const tableRef = React.useRef<HTMLTableElement | null>(null);
+    const tableMinSizes = React.useRef<null | number[]>(null);
```

Comment:
Why it could not be a `number[]` without `null`?

Reply:
So that we can identify the initial state